raritan/tests/test_decorators.py

```diff
@@ -230,10 +228,7 @@ def test_input_nonoptional_messages() -> None:
         AssertionError: If the expected log messages are not found in the captured output.
     """
     with console.capture() as capture:  # Place console capture context manager here
-        try:
-            get_missing_nonoptional_file()
-        except Exception as e:
-            error(f"Error occurred: {e}")  # Log the exception using the error() function
+        get_missing_nonoptional_file()
```

Comment:
We have a little bit of a mixed strategy in the codebase right now. The decorators sometimes raise exceptions (like when a file is missing) and sometimes just write an error message and exit(1). I think that was the case because it allowed for some more specific stack tracing to happen and maybe made testing easier.

 It felt to me like it would be better to have a consistent strategy and to add a general testing behavior. So, I moved the decorators to using the `logging.error(...)` helper consistently instead of exceptions. All the handler calls within the decorators should be wrapped in try/excepts now with logging.error calls. The exit(1) status may be bypassed with context.

Due to this work the try/excepts in the tests were no longer needed, as there should not be an exception raised. The test will still fail if one does occur. Long winded answer haha.

Reply:
Ahhh okay, thank you for explaining!